### ISSUE TITLE
stack overflow in task sdlog

### ISSUE THREAD

**Issue by moscito1010** [stuartpittaway/diyBMSv4ESP32#191]
**Describe the bug**

Controller from the Patreon Deal
Version: 2023-03-17
Precomp Image

Is it possible that the SD card is not recognized in the current controller firmware? There are also problems with communication when I want to use the "storage & logging" option

![image](https://user-images.githubusercontent.com/75945899/228042373-220e84c9-519a-4679-8cec-242ea8d2d85a.png)

If I use the firmware from 2023-03-01 there is no error and the SD card is recognized and used.

**Comment by stuartpittaway** [comment]
Hi, thanks for the detailed logs.

ESP32 should support up to 4GB of capacity - anything larger than that should still work, it just won't be "seen".

Have you tried simply re-formatting the card again (FAT32 format) ?

The SD card library/code in the controller gets updated automatically with the "core" of the ESP32/Arduino so I don't specifically change anything - although this does look like a potential bug.

**Comment by moscito1010** [comment]
I just can't get around to it, I'll try on Sunday.

**Comment by moscito1010** [comment]
well, but now....
Today I had some time and let the Raspi write it down.
putty1.log is without SD card
putty2.log is with inserted SD card

The following fact is striking.
I supply the EPS32 controller with 5v from a switching power supply via the terminal block and not via USB.
To log I have to plug a USB cable into the USB port of the ESP32.
As soon as I operate the controller without an SD card AND use an additional power source (the USB cable), does the controller run more stably?!?
With the SD card inserted, the controller is unusable and keeps restarting or freezing.
[putty1.log](https://github.com/stuartpittaway/diyBMSv4ESP32/files/11139058/putty1.log)
[putty2.log](https://github.com/stuartpittaway/diyBMSv4ESP32/files/11139059/putty2.log)

**Comment by stuartpittaway** [comment]
In the first log, there is a message Found TCA6416A, which is correct.

However, in the second log, you have hundreds of errors in the log files like this....

HAL_ESP32::WriteTCA9534APWROutputState()

This indicates the 6416A chip was not found when the SD card was inserted.

Can you upload a photo of the controller, particularly the version number.

**Comment by moscito1010** [comment]
This is the controller from the Patreon campaign. You sent it to me in the mail. Purple PCB

**Comment by stuartpittaway** [comment]
Ok, is this the one with the extra wire soldered onto the reverse?

**Comment by moscito1010** [comment]
nein

**Comment by stuartpittaway** [comment]
Reviewing the second log file, the message ```***ERROR*** A stack overflow in task sdlog has been detected.``` has been reported.

**Comment by stuartpittaway** [comment]
The sdlog overflow error should have already been fixed by commit 21d4879ab7d5aae70540305a8ea67d19289c856e

This is part of the next release.

**Comment by stuartpittaway** [comment]
The release was pushed up last week - can you test and report back please?

**Comment by moscito1010** [comment]
Sorry my response is so slow but I have so much work to do.
Today I had a few minutes and brought one of the two battery storage systems to my workshop.
I flashed the new firmware to the ESP32 and screwed the battery pack back on with 16x 1.9f modules
The controller still crashes as soon as I tap the screen, that's all I can tell. The number of O.O.S errors has dropped to ZERO. But there are only 16 and not 32 modules.

Tonight I will prepare Zen Buddhist 16 pieces 4.21 modules to use in the test. Just to confirm the 1.9f modules as a possible bug. That's only possible with a direct comparison.

**Comment by stuartpittaway** [comment]
If you don't need it, can I suggest removing the TFT display? I've got 3 running at home and none of those crash on touch, could be a faulty esp32, a soldering issue or a faulty TFT

**Comment by stuartpittaway** [comment]
How's the system operating @moscito1010  ?

**Comment by moscito1010** [comment]
Not Stuart at the moment. I am using a Battrium BMS until I get the 4.50 modules from JLCPCB. I destroyed or gave away all the 1.9f modules and I didn't have enough of the other modules. The 4.50 modules will arrive next week. Unfortunately no attinys were available, I got them from another source. When I have the boards I bring the ICś and the boards to a service provider and have the attiny's soldered on. They're too small for me, I can't solder them without a camera and magnification. I will probably have to buy helpful technology if I want to solder it myself in the future :-)
